src/sagemaker/jumpstart/utils.py

```diff
@@ -1641,8 +1641,14 @@ def remove_env_var_from_estimator_kwargs_if_accept_eula_present(
         init_kwargs (dict): Dictionary of kwargs when Estimator is instantiated.
         accept_eula (Optional[bool]): Whether or not the EULA was accepted, optionally passed in to Estimator.fit().
     """
-    if accept_eula is not None and init_kwargs["environment"]:
-        del init_kwargs["environment"][constants.SAGEMAKER_GATED_MODEL_S3_URI_TRAINING_ENV_VAR_KEY]
+    if accept_eula is not None and init_kwargs.get("environment") is not None:
+        if (
```

Comment:
nit: you can do `init_kwargs["environment"].pop(constants.SAGEMAKER_GATED_MODEL_S3_URI_TRAINING_ENV_VAR_KEY, None)`